### PR TITLE
Introduce template-based ReAct agents and Hybrid MoE strategy

### PR DESCRIPTION
**Pull request by JohnRichard4096**
## Summary by Sourcery

Introduce a new template-based ReAct agent architecture with shared core logic, add specialized Hybrid and NoAction strategies, extend lifecycle hooks and security hooks, and update protocol handling, adapters, tests, and documentation accordingly.

New Features:
- Add BaseReActAgentStrategy as a shared abstract template-method base for ReAct-style agent strategies.
- Introduce HybridReActAgentStrategy optimized for MoE models using XML-tagged tool call/result embedding into conversation context.
- Add NoActionAgentStrategy as a workflow strategy that deliberately skips tool execution.
- Expose an on_post_process lifecycle hook for all strategy categories to support post-execution context modifications.
- Add stricter message filtering for AnthropicAdapter to drop invalid assistant/tool messages before API calls.

Bug Fixes:
- Fix incorrect markdown generation for image messages by encoding raw bytes instead of decoding as UTF-8.
- Ensure ChatObject strategy execution always returns from _run_strategy and invokes on_post_process on successful completion.
- Prevent Anthropic API calls from including invalid assistant/tool messages by filtering them out in the adapter.
- Tighten MCP client and MultiClientManager behavior when already connected or misconfigured, raising clear errors.

Enhancements:
- Refactor ReActAgentStrategy to reuse shared execution flow, improve loop detection, stop handling, and error handling via the new base class.
- Improve cookie security and post-completion hooks documentation and API reference, clarifying built-in hooks and lifecycle behavior.
- Clarify and expand agent strategy, built-in agent system, and lifecycle documentation in both English and Chinese, including new strategy selection guidance.
- Improve image message handling by correctly base64-encoding raw image bytes in markdown output and adding __str__ to RawMessageContent for consistency.
- Update docs navigation to include new strategy and client classes in the API reference and fix Jinja2 template doc links.

Build:
- Add helper shell scripts to initialize a uv-based virtual environment and run the test suite with coverage.

Documentation:
- Document BaseReActAgentStrategy, HybridReActAgentStrategy, and NoActionAgentStrategy in the API reference (EN/zh).
- Expand agent strategy guides (EN/zh) with template-method architecture, built-in strategies, examples, and post-process hook usage.
- Enhance built-in adapter and agent system docs (EN/zh) with Anthropic message filtering, security considerations, and strategy selection guidelines.
- Add lifecycle documentation for on_post_process and other strategy methods in the function-implementation guides (EN/zh).

Tests:
- Add comprehensive tests for HybridReActAgentStrategy, BaseReActAgentStrategy loop handling, and differences from ReActAgentStrategy.
- Add protocol-layer tests for message content types, AdapterManager behavior, and ModelAdapter abstract method enforcement.
- Add hook tests for cookie safety and post-completion hooks.
- Extend MCP client and MultiClientManager tests to cover error paths and edge cases.
- Adjust existing tool manager tests to use unique tool names and avoid cross-test interference.

Chores:
- Bump project version from 0.7.2 to 0.7.3.

### CODE REVIEW

File: src/amrita_core/builtins/agent.py [hunk 1]
```diff
@@ -27,31 +33,80 @@
...
     agent_last_step: str | None = None
     call_count = 1
     tools: list[Any]
-    origin_msg = ""
+    origin_msg: str = ""
+    origin_instruction: str = ""
     reasoning_pc = 0
+    _suggested_stop: bool = False  # Flag to switch tool_choice from required to auto
+    _reasoning_template = Template(""""
+    Please analyze the task requirements based on the user input above,summarize the current step's purpose and reasons, and execute accordingly.
```
[the remark this answers is not in the document]
Comment (JohnRichard4096) [reply src/amrita_core/builtins/agent.py]: It's now defined in consts.py
